docs/src/docs/asciidoc/gradle-plugin-quickstart.adoc

```diff
@@ -0,0 +1,467 @@
++
+The next steps demonstrate what you should do to enable the
+https://graalvm.github.io/native-build-tools/latest/gradle-plugin.html[Gradle Plugin for GraalVM Native Image].
+. Register the plugin. Add the following to
+`plugins` section of your project’s _build.gradle_ file:
+
+[source,groovy,subs="verbatim,attributes", role="multi-language-sample"]
+----
+plugins {
+  // ...
+
+  // Apply GraalVM Native Image plugin
+  id 'org.graalvm.buildtools.native' version '{gradle-plugin-version}'
+}
+----
+
+[source,kotlin,subs="verbatim,attributes",role="multi-language-sample"]
+----
+plugins {
+  // ...
+
+  // Apply GraalVM Native Image plugin
+  id("org.graalvm.buildtools.native") version "{gradle-plugin-version}"
+}
+----
+
+The plugin discovers which JAR files it needs to pass to the
+`native-image` builder and what the executable main class should be.
+. The plugin is not yet available on the Gradle Plugin Portal, so declare an additional plugin repository. Open the _settings.gradle_ file and replace the default content with this:
+
+[source,groovy,subs="verbatim,attributes", role="multi-language-sample"]
+----
+pluginManagement {
+    repositories {
+        mavenCentral()
+        gradlePluginPortal()
+    }
+}
+
+rootProject.name = 'fortune-parent'
+include('fortune')
+----
+
+[source,kotlin,subs="verbatim,attributes", role="multi-language-sample"]
+----
+pluginManagement {
+    repositories {
+        mavenCentral()
+        gradlePluginPortal()
+    }
+}
+
+rootProject.name = "fortune-parent"
+include("fortune")
+----
+
+Note that the `pluginManagement {}` block must appear before any other statements in the file.
+
+[[build-a-native-executable-with-resources-autodetection]]
+== Build a Native Executable with Resources Autodetection
+
+You can already build a native executable by running
+`./gradlew nativeCompile` or run it directly by invoking
+`./gradlew nativeRun`. However, at this stage, running the native executable will fail because this application requires additional metadata: you need to provide it with a list of resources to load.
+
+. Instruct the plugin to automatically detect resources to be included in the native executable. Add this to your `build.gradle` file:
+
+[source,groovy,subs="verbatim,attributes", role="multi-language-sample"]
+----
+graalvmNative {
+    binaries.all {
+        resources.autodetect()
+    }
+    toolchainDetection = false
+}
+----
+
+[source,kotlin,subs="verbatim,attributes", role="multi-language-sample"]
+----
+graalvmNative {
+    binaries.all {
+        resources.autodetect()
+    }
+    toolchainDetection.set(false)
+}
+----
+
+Another thing to note here: the plugin may not be able to properly detect the GraalVM installation, because of limitations in Gradle. By default, the plugin selects a Java 11 GraalVM Community Edition. If you want to use GraalVM Enterprise, or a particular version of GraalVM and Java, you need to explicitly tell this in plugin's configuration. For example:
```

Comment:
We are no longer going to ship new JDK11-based GraalVM releases. Can this be bumped to 20?

Reply:
@fniephaus, do you mean in the future perspective? CC @melix

Reply:
Yes, or are we going to remember to update this in ~4 months?

Reply:
I submitted GR-43393 as a reminder.

Reply:
Note that the plugin needs to support _more_ than the shiniest and latest GraalVM. In particular, it's used by the Micronaut and Spring plugins. Micronaut still supports Java 8 as of today (it will move to Java 17 for Micronaut 4). So we _can't_ update the baseline that easily. You may say "wait, you say it supports Java 8, but GraalVM needs 11". That's a tradeoff we made, knowing that GraalVM no longer supports 11. As for GraalVM no longer supporting older releases (11, 17, and 20 in the future), we have made no decision yet.